src/reuse/modules/mobile/device.ts

```diff
@@ -14,42 +18,342 @@ export class Device {
     }
+    return isAppInstalledInDevice;
   }
 
   /**
    * @function installApp
    * @memberof mobile.device
    * @description Install the appropriate app based on the platform the test is being executed on.
    * @param {string} appPath - Path of the app(.apk, .ipa)
+   * @returns {Promise<void>}
    * @example
    * await mobile.device.installApp("/path/to/your/app.apk");
    * await mobile.device.installApp("/path/to/your/app.ipa");
    */
   async installApp(appPath: string): Promise<void> {
     const vl = this.vlf.initLog(this.installApp);
-    const platform: String = await browser.capabilities.platformName;
+
+    const platform: string = await browser.capabilities.platformName;
     try {
       vl.log(`Installing ${platform.toLowerCase()} app...`);
       if (["android", "ios"].includes(platform.toLowerCase().trim())) {
         await browser.installApp(appPath);
         vl.log(`${platform.toLowerCase()} app installed successfully.`);
+      } else {
+        console.error(`Unsupported platform ${platform.toLowerCase().trim()} while installing app`);
+      }
+    } catch (error) {
+      this.ErrorHandler.logException(error, `Error: Failed to installing the app: ${platform.toLowerCase().trim()}`, true);
+    }
+  }
+
+  /**
+   * @function switchToContext
+   * @memberof mobile.device
+   * @description Switch to the specified( WEBVIEW | NATIVE_APP ) context if available.
+   * @param {string} targetContext The name of the target context (default: 'WEBVIEW').
+   * @param {number} timeout Maximum time to wait for the web context to appear (default: 5000ms).
+   * @returns {Promise<boolean>} Returns 'true' if the context is successfully switched, otherwise 'false'.
+   * @example
+   * await mobile.device.switchToContext();
+   * await mobile.device.switchToContext("NATIVE_APP", 1000);
+   */
+  async switchToContext(targetContext: string = "WEBVIEW", timeout: number = 5000): Promise<boolean> {
+    const vl = this.vlf.initLog(this.switchToContext);
+
+    try {
+      const isWebContextAvailable = await browser.waitUntil(
+        async () => {
+          // Get all available contexts
+          const contexts: string[] = await browser.getContexts();
+          return contexts.some((context) => context.includes(targetContext));
+        },
+        {
+          timeout,
+          timeoutMsg: `Context "${targetContext}" not found within ${timeout}ms`
+        }
+      );
+
+      if (isWebContextAvailable) {
+        const contexts: string[] = await browser.getContexts();
+        // Find and switch to the target context
+        for (const context of contexts) {
+          if (context.includes(targetContext)) {
+            await browser.switchContext(context);
+            vl.log(`Switched to ${context} context successfully...`);
+            console.log(`Switched to ${context} context successfully...`);
+            return true;
+          }
+        }
```

Comment:
@vlad9719, I have updated the logic with new code based on your suggestion, as the context is not changing to the target value because you are passing it as a boolean value.

Reply:
its updated with new suggestions